src/tyr/route_serializer_osrm.cc

```diff
@@ -159,27 +159,49 @@ OSRM output is described in: http://project-osrm.org/docs/v5.5.1/api/
 */
 
 // Add OSRM route summary information: distance, duration
-void route_summary(json::MapPtr& route,
-                   const google::protobuf::RepeatedPtrField<valhalla::DirectionsLeg>& legs,
-                   bool imperial) {
+void route_summary(json::MapPtr& route, const valhalla::Api& api, bool imperial, int route_index) {
   // Compute total distance and duration
-  double duration = 0.0f;
-  double distance = 0.0f;
-  for (const auto& leg : legs) {
+  double duration = 0;
+  double distance = 0;
+  double weight = 0;
+  auto leg_itr = api.trip().routes(route_index).legs().begin();
+  std::vector<std::pair<double, double>> recosts(api.options().recostings_size(), {0, 0});
+  for (const auto& leg : api.directions().routes(route_index).legs()) {
     distance += leg.summary().length();
     duration += leg.summary().time();
+    weight += leg_itr->node().rbegin()->cost().elapsed_cost().cost();
+    for (int i = 0; i < leg_itr->node().rbegin()->recosts_size(); ++i) {
+      const auto& recost = leg_itr->node().rbegin()->recosts(i);
+      if (!recost.has_elapsed_cost() || recosts[i].first < 0) {
+        recosts[i].first = -1;
+        recosts[i].second = -1;
+      } else {
+        recosts[i].first += recost.elapsed_cost().seconds();
+        recosts[i].second += recost.elapsed_cost().cost();
+      }
+    }
```

Comment:
lots of this all through the serializers. at the route, leg, step and intersection all have this info though intersection does not have the recosting yet